kernel: add Microchip ENC28J60 SPI Ethernet controller module

AlbrechtL opened the pull request:
The ENC28J60 is a 10Mbit/s half-duplex Ethernet controller interfaced via SPI. It achieves real-world bandwidth up to 5Mbit/s on devices like the Raspberry Pi Zero due SPI limits. Commonly used with Raspberry Pi Zero boards for wired network connectivity.

Noltari reviewed (changes requested):
A proper commit title and description would be:
```
kernel: add Microchip ENC28J60 SPI Ethernet controller module

The ENC28J60 is a 10 Mbps half-duplex Ethernet controller interfaced via SPI.
It achieves real-world bandwidth up to 5Mbit/s on devices like the RPi Zero due
to SPI limits.
Commonly used with Raspberry Pi Zero boards for wired network connectivity.

Signed-off-by: Albrecht Lohofener <albrechtloh@gmx.de>
```

Respect the empty lines.

AlbrechtL committed:
kernel: add Microchip ENC28J60 SPI Ethernet controller module

The ENC28J60 is a 10 Mbps half-duplex Ethernet controller interfaced via SPI.
It achieves real-world bandwidth up to 5Mbit/s on devices like the RPi Zero due
to SPI limits.
Commonly used with Raspberry Pi Zero boards for wired network connectivity.

Signed-off-by: Albrecht Lohofener <albrechtloh@gmx.de>
Link: https://github.com/openwrt/openwrt/pull/19048
Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

AlbrechtL commented on package/kernel/linux/modules/netdevices.mk:
```diff
@@ -2125,3 +2125,18 @@ define KernelPackage/amazon-ena/description
+  SUBMENU:=$(NETWORK_DEVICES_MENU)
+  TITLE:=Microchip ENC28J60 SPI Ethernet chip
+  KCONFIG:=CONFIG_ENC28J60 \
+  CONFIG_ENC28J60_WRITEVERIFY=n
```
Done in https://github.com/openwrt/openwrt/pull/19048/commits/2bcae25e35fad245b16ea1579c8e92af87b987dc.

AlbrechtL commented on package/kernel/linux/modules/netdevices.mk:
```diff
@@ -2125,3 +2125,18 @@ define KernelPackage/amazon-ena/description
+
+define KernelPackage/enc28j60
+  SUBMENU:=$(NETWORK_DEVICES_MENU)
+  TITLE:=Microchip ENC28J60 SPI Ethernet chip
```
Ok. I added `driver` instead in https://github.com/openwrt/openwrt/pull/19048/commits/2bcae25e35fad245b16ea1579c8e92af87b987dc.

Noltari commented:
> ```
> The best code update.
>  
> This is the best piece of code I have ever submitted.
> Signed-off-by: John Doe <John.Doe@test.com>
> ```
> 
> Maybe the wiki needs to be updated ;-).

Done!
